### PR TITLE
School Library Basic UI Completed

### PR DESCRIPTION
**Pull request by ometman**
I have completed the basic UI of the project on branch 'basic-ui':

**The console behaves the same way as in the video >> [Watch](https://www.youtube.com/watch?v=vkkgrhD6aXQ)**

Created app.rb file that as the entry-point, with the following methods:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain Person).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person id.

**Created a file main.rb that defines the entry point, with a method **main** invoked at the end of the file:**
  - It presents the user with a list of options to perform.
  - It permits the user to choose an option.
  - In some cases, it ask for parameters for the option.
  - It also has and exit or quitting option.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 class Teacher < Person
   attr_reader :specialization
 
-  def initialize(age, specialization, name = 'UNKNOWN')
+  def initialize(age, specialization, name = 'UNKNOWN', parent_permission = true)
     super(age, name, parent_permission)
     @specialization = specialization
   end
```
Comment (Munsa1) [inline teacher.rb:9]: - [ ] There is a problem listing the `Teacher`, the teachers name is being swaped for the `Specialization`, kindly fix this.


Teacher and Student created properly
image

Teacher's name being listed as `Tech` when user selects option 2
image


#### Also note that when listing people their ID's are not being listed making it difficult to list rentals using User ID. Kindly fix this.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      person_type = person.is_a?(Student) ? 'Student' : 'Teacher'
+      generated_id = person.id
+      age = person.age if person.respond_to?(:age)
+      puts "#{index}) [#{person_type}], Name: #{person.name}, ID: #{generated_id}, Age: #{age}"
+    end
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
```
Comment (Munsa1) [inline app.rb:84]: - [ ] The app should be able to create and display rentals created by the user, kindly implement this.

Rental Created
image


image



Listing rentals using ID giving nothing
image


### Video showing errors
[cinnamon-2023-11-25T213636+0200.webm](https://github.com/ometman/oop-school-library/assets/81932478/835b3346-14c3-43e7-a975-79456570ca8c)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @ometman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Nice UI Design :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Consider adding your name to the MIT License agreement since this is your project

image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 1 above)
Comment (ometman) [reply teacher.rb:9]: You're right. Thanks. Fixed 😄

**Review by prolajumokeoni — APPROVED** [review]
Hi @ometman 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - UI updated✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**